sql/src/main/java/io/crate/metadata/doc/DocSchemaInfo.java

```diff
@@ -179,6 +176,33 @@ public TableInfo getTableInfo(String name) {
         return tables;
     }
 
+    private TableInfo getViewInfo(String name) {
+        try {
+            return viewInfoFactory.create(new TableIdent(schemaName, name), clusterService.state());
+        } catch (ResourceUnknownException e) {
+            return null;
+        }
+    }
+
+    private Collection<String> viewNames() {
+        Set<String> views = new HashSet<>();
+        ViewsMetaData viewMetaData = clusterService.state().metaData().custom(ViewsMetaData.TYPE);
+        if (viewMetaData == null) {
+            return Collections.emptySet();
+        }
+        applyFiltersAndAdd(StreamSupport.stream(viewMetaData.names().spliterator(), false), views);
+        return views;
+    }
+
+    private void applyFiltersAndAdd(Stream<String> stream, Set<String> target) {
```

Comment:
In case you want to keep the method I think it could be static and the first param could be renamed to `indexNames`